src/v/storage/file_sanitizer.h

```diff
@@ -99,9 +103,14 @@ class file_io_sanitizer final : public ss::file_impl {
           ssx::sformat(
             "ss::future<size_t>::write_dma(pos:{}, *void, len:{})", pos, len),
           maybe_inject_failure(failable_op_type::write)
-            .then([this, pos, buffer, len, pc]() {
-                return get_file_impl(_file)->write_dma(pos, buffer, len, pc);
             .then([this, pos, buffer, len, &pc]() {
+                return get_file_impl(_file)
+                  ->write_dma(pos, buffer, len, pc)
+                  .finally([this]() {
```

Comment:
There is some cost to having the `finally(` part+lambda if we are not doing failure injection -- can we somehow make this an `unlikely()` conditional and do something like:

`auto f = write_dma()
if unlikely(failure injection) {
return f.finally..
} else {
return 
}
```
?

Reply:
Good idea. Yes, we can. I'll update the PR.